### PR TITLE
Recover test_scalar_hoisting.

### PR DESCRIPTION
**Pull request by wujingyue**
I'm not sure why it wasn't added or was removed from CMakeLists.txt.

### CODE REVIEW

File: test/test_scalar_hoisting.cpp [hunk 1]
```diff
@@ -300,33 +300,33 @@ TEST_F(ScalarHoistTest, IndexHoist3) {
   const std::string expected_kernel = R"(
 __global__ void CUDAGeneratedKernel(Tensor<float, 2, 2> T0, Tensor<float, 2, 2> T2) {
   nvfuser_index_t i0;
-  i0 = ((nvfuser_index_t)threadIdx.x) + (256 * ((nvfuser_index_t)blockIdx.x));
+  i0 = ((nvfuser_index_t)threadIdx.x) + (256LL * ((nvfuser_index_t)blockIdx.x));
```
Comment (wujingyue) [inline test/test_scalar_hoisting.cpp:303]: We'd better get rid of this. It tests implementation not behavior...

**Comment by naoyam** [comment]
Looks like the test file was added in #838 but the CMake file was not updated. CC: @zasdfgbnm

File: test/test_scalar_hoisting.cpp (hunk 1 above)
Comment (naoyam) [reply test/test_scalar_hoisting.cpp:303]: The intention here is to verify hoisting of a scalar is indeed performed, so the literal matching is sufficient but not necessary. Some more intelligent matching would reduce false alarms, but we shouldn't just remove this check.

Comment (wujingyue) [reply test/test_scalar_hoisting.cpp:303]: Sure. I'll leave this test case as is, but hope someone will fix it later.

**Comment by wujingyue** [comment]
!build

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**